src/textual/widget_navigation.py

```diff
@@ -0,0 +1,192 @@
+"""
```

Comment:
Does the Textual app developer need to use this code?

I would guess not. I think this should be private, and we don't need to expose it in the api docs.

Reply:
[In a previous comment](https://github.com/Textualize/textual/pull/3912#discussion_r1439489659), Dave pointed out that if devs want to build their own widgets with sub-widgets that can be navigated with the keyboard, this might come in handy.
I hadn't thought of that and I thought this would be useful and general enough to be helpful to those.

If you really don't agree, then I'm not too fussed about it and I can make it private. 🤷

Reply:
I think we should make it private for now. Better to start private then make it public than vice versa.

Reply:
Sure.

Reply:
bfd413b.